types/index.d.ts

```diff
@@ -1301,6 +1301,12 @@ declare module 'mongoose' {
     /** The timeseries option to use when creating the model's collection. */
     timeseries?: mongodb.TimeSeriesCollectionOptions;
 
+    /** The number of seconds after which a document in a timeseries collection expires. */
+    expireAfterSeconds?: number;
+
+    /** The time after which a document in a timeseries collection expires. */
+    expires?: number | string;
```

Comment:
Could make this even safer by using the [StringValue  type provided by the ms lib](https://github.com/vercel/ms/blob/master/src/index.ts#L44):

```ts
import type { StringValue as MsStringValue } from 'ms';

// ...

/** The time after which a document in a timeseries collection expires. */
expires?: number | MsStringValue;
```

Reply:
That ist good to know. @vkarpov15 should give the Feedback how he wants to set the expiration and then I would also fix that with the last commit.

Reply:
I checked it. ms does not provide typings...

Reply:
Indeed, types will be available in [version 3](https://github.com/vercel/ms#typescript-support).